### PR TITLE
Droth 1155 strange timestamp referring to dr1 conversion on asset added in 2017

### PR DESCRIPTION
**Pull request by MAC-CGI**
Droth 1155 strange timestamp referring to dr1 conversion on asset added in 2017

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala [hunk 1]
```diff
@@ -25,47 +24,50 @@ class RoadWidthService(roadLinkServiceImpl: RoadLinkService, eventBusImpl: Digir
...
-                                  adjustedMValues = Seq.empty[MValueAdjustment],
-                                  adjustedSideCodes = Seq.empty[SideCodeAdjustment])
+      val initChangeSet: ChangeSet = ChangeSet(droppedAssetIds = Set.empty[Long],
+                                    expiredAssetIds = existingAssets.filter(asset => removedLinkIds.contains(asset.linkId)).map(_.id).toSet.filterNot(_ == 0L),
+                                    adjustedMValues = Seq.empty[MValueAdjustment],
+                                    adjustedSideCodes = Seq.empty[SideCodeAdjustment])
 
-    val combinedAssets = assetsOnChangedLinks.filterNot(a => expiredIds.contains(a.id)) ++ newRoadWidthAssets
+      val (projectedAssets, changedSetProjected) = fillNewRoadLinksWithPreviousAssetsData(projectableTargetRoadLinks,
+        assetsOnChangedLinks, assetsOnChangedLinks, changes, initChangeSet)
```
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala]: We are sending the same Sequence (assetsOnChangedLinks) on assetsToUpdate and currentAssets parameters of fillNewRoadLinksWithPreviousAssetsData function.
Maybe we can send an empty sequence to one of them.

**Comment by oskarmalmstrom** [comment]
ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala [hunk 2]
```diff
@@ -130,8 +134,16 @@ class RoadWidthService(roadLinkServiceImpl: RoadLinkService, eventBusImpl: Digir
...
+        val id = (linearAsset.createdBy, linearAsset.createdDateTime) match {
+          case (Some(createdBy), Some(createdDateTime)) =>
+            dao.createLinearAsset(linearAsset.typeId, linearAsset.linkId, linearAsset.expired, linearAsset.sideCode,
+              Measures(linearAsset.startMeasure, linearAsset.endMeasure), linearAsset.modifiedBy.getOrElse(LinearAssetTypes.VvhGenerated), linearAsset.vvhTimeStamp,
+              getLinkSource(roadLinks.find(_.linkId == linearAsset.linkId)), fromUpdate = true, Some(createdBy), Some(createdDateTime), linearAsset.verifiedBy, linearAsset.verifiedDate)
+          case _ =>
+            dao.createLinearAsset(linearAsset.typeId, linearAsset.linkId, linearAsset.expired, linearAsset.sideCode,
+              Measures(linearAsset.startMeasure, linearAsset.endMeasure), linearAsset.createdBy.getOrElse(LinearAssetTypes.VvhGenerated), linearAsset.vvhTimeStamp,
+              getLinkSource(roadLinks.find(_.linkId == linearAsset.linkId)))
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala]: Shouldn't this create have the verifiedBy and verifiedDate?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala (hunk 1 above)
Comment (MAC-CGI) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/RoadWidthService.scala]: In this case we didn't have any difference between 'assetsToUpdate' and 'currentAssets' . For that reason 'assetsOnChangedLinks' is sent twice.

**Pull request merged**